src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -1888,12 +1888,28 @@ void WriteSingleCharChild(RegexNode child, bool includeDescription = true)
                         continue;
                     }
 
-                    EmitNode(node.Child(i), i + 1 < childCount ? node.Child(i + 1) : subsequent, emitLengthChecksIfRequired: emitLengthChecksIfRequired);
+                    EmitNode(node.Child(i), GetSubsequent(i, node, subsequent), emitLengthChecksIfRequired: emitLengthChecksIfRequired);
                     if (i < childCount - 1)
                     {
                         writer.WriteLine();
                     }
                 }
+
+                // Gets the node to treat as the subsequent one to node.Child(index)
+                static RegexNode? GetSubsequent(int index, RegexNode node, RegexNode? subsequent)
```

Comment:
Very small NIT so feel free to not fix, but seems like the right name for this method should be something like `GetSubsequentOrDefault` and the last parameter should probably be renamed to be `@default` or something like that.

Reply:
Sure, I need to resolve conflicts anyway :)